test/complex.jl

```diff
@@ -76,14 +76,20 @@ fs_C_to_C_non_holomorphic = (conj,
                              z->im*abs2(z),
                              z->z'z,
                              z->conj(z)*z^2,
+                             z->imag(z)^2+real(sin(z))^3*1im,
                              )
 @testset "C->C non-holomorphic" begin
-    for f in (fs_C_to_C_holomorphic...,fs_C_to_C_holomorphic...)
+    for f in (fs_C_to_C_holomorphic...,fs_C_to_C_non_holomorphic...)
```

Comment:
20 lines up there's another loop over only the holomorphic list. By joining the two lists here, it looks like the intention may have been to delete the other loop?

Reply:
Indeed, I suppose that includes an assertion to check the function is holomorphic but thats the only difference. We could add that to this loop conditional on it being a function in the holomorphic list?

Reply:
Having thought a little more, perhaps its best to keep it as two distinct loops and remove it here? They are somewhat distinct, and it could be good to add a check that Zygote produces a holomorphic gradient (i.e. compare the gradients of the real and imag outputs) for holomorphic functions, similar to how this is done with finite-diff to check the function actually is holomorphic.

Reply:
Done!